Xsuaa support in cds-feature-identity (#431)
cap-js/docs: java/security.md

### Configure XSUAA and IAS Authentication { #xsuaa-ias}
To enable your application for XSUAA or IAS-authentication we recommend to use the `cds-starter-cloudfoundry` or the `cds-starter-k8s` starter bundle, which covers all required dependencies.

:::details Individual Dependencies
These are the individual dependencies that can be explicitly added in the `pom.xml` file of your service:
   * `com.sap.cloud.security:resourceserver-security-spring-boot-starter` that brings [spring-security library](https://github.com/SAP/cloud-security-services-integration-library/tree/main/spring-security)
   * `org.springframework.boot:spring-boot-starter-security`
   * `cds-feature-identity`

> :::

Reviewer: Restructured it, so that the bundles are mentioned first and individuals dependencies are in a collapsed section:
![image](https://github.com/cap-js/docs/assets/32062331/7be410af-e358-4cac-83d4-b24f78b74a95)

User requests with invalid authentication need to be rejected as soon as possible, to limit the resource impact to a minimum. Ideally, authentication is one of the first steps when processing a request. This is one reason why it's not an integral part of the CAP runtime and needs to be configured on application framework level. In addition, CAP Java is based on a [modular architecture](architecture#modular_architecture) and allows flexible configuration of the authentication method. For productive scenarios, [XSUAA and IAS](#xsuaa-ias) authentication is supported out of the box, but a [custom authentication](#custom-authentication) can be configured as well. For the local development and test scenario, there's a built-in [mock user](#mock-users) support.

### Configure XSUAA and IAS Authentication { #xsuaa-ias}
To enable your application for XSUAA or IAS-authentication the following dependencies need to be set:
   * `com.sap.cloud.security:resourceserver-security-spring-boot-starter` that brings [spring-security library](https://github.com/SAP/cloud-security-services-integration-library/tree/main/spring-security)
   * `org.springframework.boot:spring-boot-starter-security` 

>    * `cds-feature-identity`

Reviewer 2: We shouild point to recommended cds-starteer-bundles cf and k8 here. We don't expect customers to set the dependecies individually.

The individual dependencies can be explicitly added in the `pom.xml` file of your service.

> In addition, your application needs to be bound to corresponding service instances depending on your scenario. If your application should:

Reviewer 2: ```suggestion
In addition, your application needs to be bound to corresponding service instances depending on your scenario. If your application should...
```

Reviewer: ```suggestion
In addition, your application needs to be bound to corresponding service instances depending on your scenario. The following list describes which services needs to be bound depending on the tokens your applications should accept:
```

>    * only accept tokens issued by XSUAA, bind your application to a [XSUAA service instance](../guides/authorization#xsuaa-configuration)

Reviewer 2: ```suggestion
   * only accept tokens issued by XSUAA, bind your application to an [XSUAA service instance](../guides/authorization#xsuaa-configuration)
```

Reviewer: ```suggestion
   * only accept tokens issued by XSUAA --> bind your application to an [XSUAA service instance](../guides/authorization#xsuaa-configuration)
```

>    * only accept tokens issued by IAS, bind your application to a [IAS service instance](https://help.sap.com/docs/IDENTITY_AUTHENTICATION)

Reviewer 2: ```suggestion
   * only accept tokens issued by IAS, bind your application to an [IAS service instance](https://help.sap.com/docs/IDENTITY_AUTHENTICATION)
```

Reviewer: ```suggestion
   * only accept tokens issued by IAS --> bind your application to an [IAS service instance](https://help.sap.com/docs/IDENTITY_AUTHENTICATION)
```

>    * accept tokens issued by XSUAA and IAS, bind your application to service instances of both types.

Reviewer: ```suggestion
   * accept tokens issued by XSUAA and IAS, bind your application to service instances of both types
```

::: tip
CAP Java picks only a single binding of each type. If you have multiple XSUAA or IAS bindings, choose a specific binding with property `cds.security.xsuaa.binding`respectively `cds.security.identity.binding`.
Choose an appropriate XSUAA service plan to fit the requirements. For instance, if your service should be exposed as technical reuse service, make use of plan `broker`.
:::

### Transition from `cds-feature-xsuaa` to `cds-feature-identity`{ #transition-xsuaa-ias}
CAP also provides support for XSUAA-based authentication via the maven dependency `cds-feature-xsuaa` which is based on the [spring-xsuaa library](https://github.com/SAP/cloud-security-services-integration-library/tree/main/spring-xsuaa). As the spring-xsuaa library is deprecated, it is recommended to move to `cds-feature-identity`.

> To do so, remove existing dependencies to `cds-feature-xsuaa` and `xsuaa-spring-boot-starter` and follow the description in [Configure XSUAA and IAS Authentication]{.#xsuaa-ias}.  

Reviewer 2: ```suggestion
To do so, remove existing dependencies to `cds-feature-xsuaa` and `xsuaa-spring-boot-starter` and follow the description in [Configure XSUAA and IAS Authentication]{#xsuaa-ias}.  
```
